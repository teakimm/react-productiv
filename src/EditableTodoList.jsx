import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */



function EditableTodoList({ todos, update, remove, complete }) {

  return (
    <div>
      {todos.map(t =>
        <EditableTodo
          todo={t}
          update={update}
          remove={remove}
          complete={complete}
          key={t.id}
        />)}
    </div>
  );
}

export default EditableTodoList;
