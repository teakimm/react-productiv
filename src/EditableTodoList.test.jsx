import { describe, it, expect, test } from "vitest";

import React from "react";
import { fireEvent, render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

const testTodos = [
    {
        id: 1,
        title: "Code!",
        description: "Write some code",
        priority: 2,
    },
    {
        id: 2,
        title: "Make dinner",
        description: "Cook something healthy",
        priority: 1,
    },
    {
        id: 3,
        title: "Go to bed",
        description: "In bed by 11:15",
        priority: 3,
    },
];

it("renders without crashing", function () {
    render(<EditableTodoList todos={testTodos} />);
});

it("matches snapshot", function () {
    const { container } = render(<EditableTodoList todos={testTodos} />);
    expect(container).toMatchSnapshot();
});

it("renders editableTodos", function () {
    const result = render(<EditableTodoList todos={testTodos} />);

    expect(
        result.queryByText("Write some code")
    ).toBeInTheDocument();
    expect(
        result.queryByText("Cook something healthy")
    ).toBeInTheDocument();
    expect(
        result.queryByText("In bed by 11:15")
    ).toBeInTheDocument();
    expect(
        result.queryByText("lmao")
    ).not.toBeInTheDocument();
});