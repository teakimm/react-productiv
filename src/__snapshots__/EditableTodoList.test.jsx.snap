// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Code!
            </b>
             
            <small>
              (
              priority: 2
              )
            </small>
          </div>
          <div>
            <small>
              Write some code
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Make dinner
            </b>
             
            <small>
              (
              priority: 1
              )
            </small>
          </div>
          <div>
            <small>
              Cook something healthy
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              Go to bed
            </b>
             
            <small>
              (
              priority: 3
              )
            </small>
          </div>
          <div>
            <small>
              In bed by 11:15
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
